plugins/python/redisgears_python.c

```diff
@@ -1746,6 +1747,572 @@ static PyObject* atomicExit(PyObject *self, PyObject *args){
+    PyTensor* pyt = (PyTensor*)pyObj;
+    RedisAI_TensorFree(pyt->t);
+    Py_TYPE(pyObj)->tp_free((PyObject*)pyObj);
+}
+
+static PyTypeObject PyTensorType = {
+  PyVarObject_HEAD_INIT(NULL, 0)
+  "redisgears.PyTensor",       /* tp_name */
+  sizeof(PyTensor),          /* tp_basicsize */
+  0,                         /* tp_itemsize */
+  PyTensor_Destruct,         /* tp_dealloc */
+  0,                         /* tp_print */
+  0,                         /* tp_getattr */
+  0,                         /* tp_setattr */
+  0,                         /* tp_compare */
+  0,                         /* tp_repr */
+  0,                         /* tp_as_number */
+  0,                         /* tp_as_sequence */
+  0,                         /* tp_as_mapping */
+  0,                         /* tp_hash */
+  0,                         /* tp_call */
+  PyTensor_ToStr,            /* tp_str */
+  0,                         /* tp_getattro */
+  0,                         /* tp_setattro */
+  0,                         /* tp_as_buffer */
+  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,        /* tp_flags */
+  "PyTensor",                /* tp_doc */
+};
+
+typedef struct PyDAGRunner{
+    PyObject_HEAD
+    RAI_DAGRunCtx* dag;
+} PyDAGRunner;
+
+static PyObject* PyDAG_ToStr(PyObject *obj){
+    return PyUnicode_FromString("PyDAGRunner to str");
+}
+
+static void PyDAGRunner_Destruct(PyObject *pyObj){
+    PyDAGRunner* pyDag = (PyDAGRunner*)pyObj;
+    if (pyDag->dag) RedisAI_DAGFree(pyDag->dag);
+    Py_TYPE(pyObj)->tp_free((PyObject*)pyObj);
+}
+
+static PyTypeObject PyDAGRunnerType = {
+  PyVarObject_HEAD_INIT(NULL, 0)
+  "redisgears.PyDAGRunner",             /* tp_name */
+  sizeof(PyDAGRunner), /* tp_basicsize */
+  0,                         /* tp_itemsize */
+  PyDAGRunner_Destruct,    /* tp_dealloc */
+  0,                         /* tp_print */
+  0,                         /* tp_getattr */
+  0,                         /* tp_setattr */
+  0,                         /* tp_compare */
+  0,                         /* tp_repr */
+  0,                         /* tp_as_number */
+  0,                         /* tp_as_sequence */
+  0,                         /* tp_as_mapping */
+  0,                         /* tp_hash */
+  0,                         /* tp_call */
+  PyDAG_ToStr,                         /* tp_str */
+  0,                         /* tp_getattro */
+  0,                         /* tp_setattro */
+  0,                         /* tp_as_buffer */
+  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,        /* tp_flags */
+  "PyDAGRunner",           /* tp_doc */
+};
+
+static PyObject* createDAGRunner(PyObject *cls) {
+    verifyRedisAILoaded();
+    RAI_DAGRunCtx* dagCtx = RedisAI_DAGRunCtxCreate();
+    PyDAGRunner* pyDag = PyObject_New(PyDAGRunner, &PyDAGRunnerType);
+    pyDag->dag = dagCtx;
+
+    return (PyObject*) pyDag;
+}
+/*
+ * Methods for RedisAI DAG runner
+ */
+
+static PyObject* loadInput(PyObject *self, PyObject *args) {
```

Comment:
Maybe DAGLoadInput so we will know its a DAG function?

Reply:
Actually Dvir suggested to remove the "DAG" prefix from the methods name, to make it more "python stile". I doesn't really matter to me either way...

Reply:
The python method name and C method name do not have to be the same ... but regarding naming it's your call whatever you think it best ...

Reply:
@alonre24 I meant only the python names